src/utils/ProtocolMetrics.ts

```diff
@@ -78,52 +96,108 @@ function getSklimaSupply(transaction: Transaction): BigDecimal {
+        // Treasury LP token balance
+        treasuryKLIMABCT.tokenBalance = toDecimal(klimabctERC20.balanceOf(treasuryAddress), 18)
 
-    let mv = co2ValueDecimal
-    let rfv = co2ValueDecimal.plus(rfvLpValue)
+        // Get total LP supply and calc treasury percent
+        let total_lp = toDecimal(klimabctUNIV2.totalSupply(), 18)
+        let ownedLP = treasuryKLIMABCT.tokenBalance.div(total_lp)
+        treasuryKLIMABCT.POL = ownedLP
 
-    //log.debug("Treasury Market Value {}", [mv.toString()])
-    log.debug("Treasury RFV {}", [rfv.toString()])
-    log.debug("Treasury BCT value {}", [toDecimal(bctBalance, 18).toString()])
-    log.debug("Treasury KLIMA-BCT RFV {}", [klimabct_rfv.toString()])
+        // Percent of Carbon in LP owned by the treasury
+        treasuryKLIMABCT.carbonBalance = toDecimal(klimabctUNIV2.getReserves().value0, 18).times(ownedLP)
+        treasuryKLIMABCT.marketValue = treasuryKLIMABCT.carbonBalance.times(getBCTUSDRate())
+
+    }
+
+    treasuryKLIMABCT.save()
+
+    // BCT-USDC
+
+    let bctusdcERC20 = ERC20.bind(Address.fromString(constants.BCT_USDC_PAIR))
+    let bctusdcUNIV2 = UniswapV2Pair.bind(Address.fromString(constants.BCT_USDC_PAIR))
+
+    let treasuryBCTUSDC = loadOrCreateTreasuryAsset(transaction.timestamp, constants.BCT_USDC_PAIR)
+    if (transaction.blockNumber.gt(BigInt.fromString(constants.BCT_USDC_PAIR_BLOCK))) {
+
+        // Treasury LP token balance
+        treasuryBCTUSDC.tokenBalance = toDecimal(bctusdcERC20.balanceOf(treasuryAddress), 18)
+
+        // Get total LP supply and calc treasury percent
+        let total_lp = toDecimal(bctusdcUNIV2.totalSupply(), 18)
+        let ownedLP = treasuryBCTUSDC.tokenBalance.div(total_lp)
+        treasuryBCTUSDC.POL = ownedLP
+
+        // Percent of Carbon in LP owned by the treasury
+        treasuryBCTUSDC.carbonBalance = toDecimal(bctusdcUNIV2.getReserves().value1, 18).times(ownedLP)
+        treasuryBCTUSDC.marketValue = treasuryBCTUSDC.carbonBalance.times(getBCTUSDRate())
+
+    }
+
+    treasuryBCTUSDC.save()
 
     return [
-        mv,
-        rfv,
-        co2ValueDecimal,    // Total Carbon RFV
-        klimabct_rfv,       // KLIMA-BCT RFV
-        klimabctPOL         // KLIMA-BCT POL
+        treasuryBCT.id,
+        treasuryKLIMABCT.id,
+        treasuryBCTUSDC.id
     ]
 }
 
+function getMV_RFV(transaction: Transaction, assets: string[]): BigDecimal[] {
+
+    let totalCarbon = BigDecimal.fromString("0")
+    let totalRFV = BigDecimal.fromString("0")
+    let totalMarketValue = BigDecimal.fromString("0")
+
+    for (let i = 0; i < assets.length; i++) {
+        let assetDetail = loadOrCreateTreasuryAsset(transaction.timestamp, assets[i].substring(10))
+        totalCarbon = totalCarbon.plus(assetDetail.carbonBalance)
+        totalRFV = totalRFV.plus(assetDetail.riskFreeValue)
+        totalMarketValue = totalMarketValue.plus(assetDetail.marketValue)
+    }
+
+    //log.debug("Treasury Market Value {}", [mv.toString()])
```

Comment:
Do we want these debug logs to be left in here? Looks like one is still commented

Reply:
This is actually duplicated on line 192. Removed the commented duplicate log message.